routes.js

```diff
@@ -23,6 +25,8 @@ routes.post("/", authMiddleware, (req, res) => {
 
 routes.post('/user', createUserValidator, createUser.handle);
 
+routes.get('/user', authMiddleware, meController.handle);
```

Comment:
E qual é a rota que retorna todos os usuários?

sugestão:
POST /users (cria usuário)
GET /users (retorna coleção de usuários)
GET /users/me (alias para /users/:id, onde :id vc pega do token)
GET /users/:id (retorna usuário por id)
PUT /users/:id (upsert de usuario)
DELETE /users/:id (deleta o usuario)

Reply:
Essa call é muito boa

Reply:
Oi galera, então. to pensando em padronizar isso.
A aplicação não tem a necessidade de fato de ter uma rota para retornar todos os usuários. 
Vou atualizar a rota para o GET `user/me`, tinha até comentado isso no twitter. 

Pelo fato da aplicação não ter necessidade de retornar todos os usuários, não faz sentido eu criar essa rota né ?

Reply:
@NicolasPereira sem problemas, vc não precisa implementar as rotas as quais vc não vai usar. Mas repare que, independente se vc vai implementar uma determinada rota agora ou no futuro, elas não devem conflitar.

Repare também que eu estou sempre sugerindo `/users` (no plural). Isso é porque essas rotas representam coleções e elementos dentro dessas coleção (`/users/123`). E eu, particularmente, prefiro utilizar o plural para explicitar isso.

Nomear uma coleção no singular seria equivalente a isso:
```js
const user = [user1, user2, user3, ...]
```

Já no plural seria equivalente a isso:
```js
const users = [user1, user2, user3, ...]
```

É bem mais natural quando damos um nome no plural para um coleção/lista né?

Reply:
Entendi sim, faz sentido! Valeu